examples/md-flexible/src/configuration/YamlParser.cpp

```diff
@@ -1,386 +1,769 @@
+  std::array<double, 3> bottomLeftCorner = parseBottomLeftCorner(config, it);
+
+  CubeUniform cubeUniform(velocity, particleType, epsilon, sigma, mass, numParticles, boxLength, bottomLeftCorner);
+  return cubeUniform;
+}
+
+CubeGauss MDFlexParser::YamlParser::parseCubeGaussObject(MDFlexConfig &config, YAML::iterator &it) {
+  std::array<double, 3> velocity = parseVelocity(config, it);
+  unsigned long particleType = parseParticleType(config, it);
+  double epsilon = parseEpsilon(config, it);
+  double sigma = parseSigma(config, it);
+  double mass = parseMass(config, it);
+  size_t numParticles = parseNumParticles(config, it);
+  std::array<double, 3> boxLength = parseBoxLength(config, it);
+  std::array<double, 3> distributionMean = parseDistrMean(config, it);
+  std::array<double, 3> distributionStdDev = parseDistrStdDev(config, it);
+  std::array<double, 3> bottomLeftCorner = parseBottomLeftCorner(config, it);
+
+  CubeGauss cubeGauss(velocity, particleType, epsilon, sigma, mass, numParticles, boxLength, distributionMean,
+                      distributionStdDev, bottomLeftCorner);
+  return cubeGauss;
+}
+
+Sphere MDFlexParser::YamlParser::parseSphereObject(MDFlexConfig &config, YAML::iterator &it) {
+  std::array<double, 3> velocity = parseVelocity(config, it);
+  unsigned long particleType = parseParticleType(config, it);
+  double epsilon = parseEpsilon(config, it);
+  double sigma = parseSigma(config, it);
+  double mass = parseMass(config, it);
+  std::array<double, 3> sphereCenter = parseCenter(config, it);
+  double sphereRadius = parseRadius(config, it);
+  double particleSpacing = parseParticleSpacing(config, it);
+
+  Sphere sphere(velocity, particleType, epsilon, sigma, mass, sphereCenter, sphereRadius, particleSpacing);
+  return sphere;
+}
+
+CubeClosestPacked MDFlexParser::YamlParser::parseCubeClosestPacked(MDFlexConfig &config, YAML::iterator &it) {
+  std::array<double, 3> velocity = parseVelocity(config, it);
+  unsigned long particleType = parseParticleType(config, it);
+  double epsilon = parseEpsilon(config, it);
+  double sigma = parseSigma(config, it);
+  double mass = parseMass(config, it);
+  double particleSpacing = parseParticleSpacing(config, it);
+  std::array<double, 3> boxLength = parseBoxLength(config, it);
+  std::array<double, 3> bottomLeftCorner = parseBottomLeftCorner(config, it);
+
+  CubeClosestPacked cubeClosestPacked(velocity, particleType, epsilon, sigma, mass, particleSpacing, boxLength,
+                                      bottomLeftCorner);
+
+  return cubeClosestPacked;
+}
+
 bool MDFlexParser::YamlParser::parseYamlFile(MDFlexConfig &config) {
+  /*
+  Global variables used to print the expected input and a description of the parameter if an error occurs while
+  parsing. Yaml mark is used to identify the current line of the error.
+  */
+  std::string expected;
+  std::string description;
+  YAML::Mark m;
+  bool errorOccured = false;
```

Comment:
Do you actually need this bool? Can't we just check if `errors.empty()`?

Reply:
done [9255684](https://github.com/AutoPas/AutoPas/pull/732/commits/92556841e912239c425f4d4654d0110b6d71701e)